src/cpp/rtps/builtin/data/ParticipantProxyData.cpp

```diff
@@ -92,15 +96,8 @@ ParticipantProxyData::~ParticipantProxyData()
 {
     logInfo(RTPS_PARTICIPANT, m_guid);
 
-    for (ReaderProxyData* it : m_readers)
-    {
-        delete it;
-    }
-
-    for (WriterProxyData* it : m_writers)
-    {
-        delete it;
-    }
+    delete m_readers;
+    delete m_writers;
```

Comment:
Old behavior deleted the proxy objects held by the collections, new behavior does not do that. This implies a memory leak.

Reply:
Initially we didn't keep pointers :( later I forgot to mend this behavior.